infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -340,11 +340,11 @@ private synchronized void runDetectionTasks() {
             // Corrects out of phase epoch issues if present in the report. This method
             // performs re-sealing of all nodes if required and catchup of a layout server to
             // the current state.
-            correctWrongEpochs(pollReport);
+            Layout updatedLayout = correctWrongEpochs(pollReport, layout);
 
             Result<DetectorTask, RuntimeException> failure = Result.of(() -> {
                 if (pollReport.isCurrentLayoutSlotUnFilled()) {
-                    detectFailure(Collections.emptySet(), pollReport).join();
+                    detectFailure(updatedLayout, Collections.emptySet(), pollReport).join();
```

Comment:
The term updatedLayout is a bit misleading here. It might not have been updated if the seal fails.
Can we rename this? If not can we add a comment her explaining this?

Reply:
what name do you prefer?